news/_posts/2019-05-23-announcing-end-of-support-for-obsolete-php-versions.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Announcing end of support for obsolete PHP versions"
+subtitle: "It's time to move forward"
+date:   2019-05-23 09:10:11
+authors: [ PabloBorowicz ]
+icon: icon-compass
+tags:
+ - 1.7
+ - architecture
+ - development
+---
+
+About a year and a half ago, we announced that PrestaShop 1.7.4.0 would be [dropping support for PHP 5.4 and 5.5](http://build.prestashop.com/news/prestashop-1-7-is-moving-to-symfony-3-4-and-php-5-6/). At that time, this decision was justified by the impending end-of-maintenance for Symfony 2 and its replacement by Symfony 3, which didn't support PHP versions older than 5.5. 
+
+By late 2018, as the end-of-life of PHP 5.6 approached, a great deal of shared hosting providers started dropping support for that version on their services and upgrading their clients, forcibly or otherwise, to an officially maintained PHP 7.x version. Our metrics show that during the last year, the number of active shops running PrestaShop 1.7 on a PHP 7 or greater has increased significantly and is now over 75%.
+
+![PHP version support roadmap](/assets/images/2019/05/php-support-roadmap.png){:.text-center}
+
+_PHP version support roadmap  
+(green background: active support; orange background: security fixes only) - ([Source](https://www.php.net/supported-versions.php))_
+{:.text-center}
+
+Fast forward to 2019, the exceptionally long support period that PHP 5.6 benefited from is now over. As of May 2019, PHP is back on a shorter, three year-long version lifecycle, consisting on a new minor release every year, with two years of active support and one of security fixes only. As a result, the PHP community has started picking up speed as well, adopting new versions and dropping compatibility with older versions faster than ever.
+
+
+## PrestaShop is directly affected by this change. But how exactly?
+
+First and foremost, as you probably know, the internet is a dangerous place... and businesses are prime targets for attackers. As a merchant or shop maintainer, you need to be aware that it's dangerous to run any web service on outdated software, whether it is your operating system, web server, PrestaShop itself or the scripting language it's written in, PHP. Outdated software may contain a known vulnerability that could be exploited by an attacker to take over your shop and do serious damage to your business. Keeping your software stack up to date ensures you get the latest security fixes, and reduces the risk that your shop will be compromised.
```

Comment:
```suggestion
First and foremost, as you probably know, the internet is a dangerous place... and businesses are prime targets for attackers. As a merchant or shop maintainer, you need to be aware that it's dangerous to run any web service on outdated software, whether it is your operating system, web server, PrestaShop itself or its scripting language, PHP. Outdated software may contain a known vulnerability that could be exploited by an attacker to take over your shop and do serious damage to your business. Keeping your software stack up to date ensures you get the latest security fixes, and reduces the risk that your shop will be compromised.
```

Reply:
> PrestaShop itself or its scripting language

That would imply that PHP it's PrestaShop's own scripting language. How about this?

```suggestion
First and foremost, as you probably know, the internet is a dangerous place... and businesses are prime targets for attackers. As a merchant or shop maintainer, you need to be aware that it's dangerous to run any web service on outdated software, whether it is your operating system, web server, PrestaShop or the scripting language PrestaShop is written in, PHP. Outdated software may contain a known vulnerability that could be exploited by an attacker to take over your shop and do serious damage to your business. Keeping your software stack up to date ensures you get the latest security fixes, and reduces the risk that your shop will be compromised.

Reply:
This paragraph should be removed. It is not specific to PrestaShop, nor to PHP. It adds no value to the article; this is not about security, but about why we must drop old PHP versions to support upcoming PHP versions.

The community is currently blaming us, considering that we are using security in our communication to force 1.6 users to upgrade to the latests 1.7.x. On the forum, there has been already many hard discussions about old versions of PHP, with no consensus and people standing on their position. This article must not nurture that kind of discussions and behaviors.